### PR TITLE
docs: added flow diagram and other changes to create oauth2_client api contract

### PR DESCRIPTION
**Pull request by AdithyaSSup**
contract changes to create oauth2_client api

### CODE REVIEW

File: api_spec.yaml [hunk 1]
```diff
@@ -809,30 +810,22 @@ paths:
...
-                      $ref: '#/components/schemas/oauth2Scope'
+              type: object
+              properties:
+                redirectUris:
+                  type: array
+                  items:
+                    $ref: '#/components/schemas/oauth2RedirectUri'
+                name:
+                  type: string
+                  example: 'developement'
```
Comment (rishabhpoddar) [inline api_spec.yaml:821]: name in request example not matching response..

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: api_spec.yaml (hunk 1 above)
Comment (AdithyaSSup) [reply api_spec.yaml:821]: changes made

File: api_spec.yaml [hunk 2]
```diff
@@ -809,30 +810,22 @@ paths:
...
+                scopes:
+                  type: array
+                  items:
+                    $ref: '#/components/schemas/oauth2Scope'
+              required:
+                - redirectUris
+                - name
```
Comment (rishabhpoddar) [inline api_spec.yaml:828]: in the request add an optional boolean for `addDefaultSuperTokensScopes` (default value is true). 

Change the flow diagram accordingly to take this boolean into account.

Comment (rishabhpoddar) [reply api_spec.yaml:828]: default scopes start with `supertokens:...`

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: api_spec.yaml (hunk 2 above)
Comment (AdithyaSSup) [reply api_spec.yaml:831]: changes made to include :
- clients with no scopes
- clients with default scopes
- clients with default scopes as well as existing user-defined scopes

**Pull request merged**